oneOfWeighted.js

```diff
@@ -11,20 +11,14 @@ function oneOfWeighted(a, b) {
 
 function oneOfWeightedMain(input, samples) {
   samples = parseSamples(samples)
-  var id = hash2(input, 'oneOfWeighted')
-  var n = samples.length
-  var pRemaining = 1
-  var i = -1
-  var sample
-  var p
-
-  while (++i < n) {
-    sample = samples[i]
-    p = sample[0] / pRemaining
-    pRemaining -= p
-
-    if (flip(id, p)) {
-      return resolve(id, sample[1])
+  var id = hash(input, 'oneOfWeighted')
+  const prob = (id % 1_000_000) / 1_000_000
+
+  let cumulative = 0;
+  for (const [probability, value] of samples) {
```

Comment:
When I created this project 5 years ago, I had the brainwave of writing it only in es5 to avoid a build step. I'm not sure if that decision was the right one, but unfortunately it means for now we need to write the code as es5 for linting to pass. I'm sorry about this, I realise it is absurd to ask someone to write es5 in 2025 :)

```suggestion
  var cumulative = 0;
  for (var i = 0; i < samples.length; i++) {
    var probability = samples[i][0];
    var value = samples[i][1];
```

Reply:
alright. happy to comply.